src/main/clj/com/climate/services/aws/ec2.clj

```diff
@@ -9,19 +9,13 @@
      DescribeSpotPriceHistoryRequest
      DescribeInstancesRequest
      Filter]
-    [com.amazonaws.auth BasicAWSCredentials]
+    [com.amazonaws.auth DefaultAWSCredentialsProviderChain]
     [java.util Date GregorianCalendar]))
 
-(def ^{:dynamic true} *ec2* nil)
+(def ^:dynamic *ec2* nil)
 
-(defn ec2-client [aws-creds]
-  (AmazonEC2Client. aws-creds))
-
-; TODO All functions that use this dynamic var should have an additional fn
-;      signature where the object can be passed in explicitly
-(defn ec2
-  [creds]
-  (aws ec2-client creds))
+(defn ec2 [^DefaultAWSCredentialsProviderChain credentials]
```

Comment:
Please tag with the interface implemented, not the class you're using. It's more flexible. I think in this case, that would be AWSCredentialsProvider.